### PR TITLE
feat:  update api key validation 178

### PR DESCRIPTION
**Pull request by sonebo**
With this PR we no longer validate the format of the API key - instead we add a button to send a test request. The test button is present when you create a new team, as well as when you want to update an existing team. 

I opted to use the test request that they have on Open AI's reference docs: https://platform.openai.com/docs/api-reference/making-requests. 

Let me know what you think!

### CODE REVIEW

**Review by joakimunge — APPROVED** [review]
Nice work!

File: src/lib/utils/testOpenAiAPIKey.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    })
+    if (response.status === 200) {
+      return 'The API key is valid.'
+    } else {
+      return `The API key is invalid or has insufficient permissions. Status code: ${response.status}.`
+    }
+  } catch (error) {
+    return 'An error occurred while testing the API key.'
+  }
+}
```
Comment (brokelyn) [inline src/lib/utils/testOpenAiAPIKey.ts:23]: dont they have a test endpoint that we could use?

File: src/lib/components/TeamKeys.svelte [hunk 2]
```diff
@@ -36,7 +39,13 @@
       disabled={userTeam.role === 'MEMBER'}
       value={form?.fields?.openAiApiKey ?? team.openAiApiKey}
       errors={form?.errors?.openAiApiKey}
+      on:input={(event) => {
+        testApiKey = event.detail.target?.value
+      }}
     />
+    <div class="col-span-full">
+      <TestApiButton {testApiKey} />
+    </div>
```
Comment (brokelyn) [inline src/lib/components/TeamKeys.svelte:48]: Some svelte magic:

```suggestion
  <TestApiButton class="col-span-full" {testApiKey} />
```

and then you do inside the component `<div class="flex xyz {$$props.class}">`

**Review by brokelyn — APPROVED** [review]
Nice looks good!

File: src/lib/utils/testOpenAiAPIKey.ts (hunk 1 above)
Comment (sonebo) [reply src/lib/utils/testOpenAiAPIKey.ts:23]: I couldn't find one in their docs, so I opted for the example on their completions endpoint that they use here: https://platform.openai.com/docs/api-reference/making-requests. But let's switch it out if we find something better in the future 👍

File: src/lib/components/TeamKeys.svelte (hunk 2 above)
Comment (sonebo) [reply src/lib/components/TeamKeys.svelte:48]: Fancy!! I'll make the changes asap ✨

**Pull request merged**